extras/parse_travis_log.py

```diff
@@ -20,396 +20,894 @@
 # along with NEST.  If not, see <http://www.gnu.org/licenses/>.
 
 """
-This script parses the TravisCI output as it is generated
-by ./build.sh and outputs a shorter summary. It is hard-coded
-to the output of ./build.sh, make installcheck,
-and cmake -DCMAKE_INSTALL_PREFIX=... . Changing any of those, requires
-adapting this script.
+This python script is part of the NEST Travis CI build and test environment.
+It parses the Travis CI build log file 'build.sh.log' (The name is hard-wired
+in '.travis.yml'.) and creates the 'NEST Travis CI Build Summary'.
+
+NOTE: Please note that the parsing process is coupled to shell script
+      'build.sh' and relies on the message numbers "MSGBLDnnnn'.
+      It does not rely on the messages texts itself except for file names.
 """
 
-INDENT = "  "
 
+def is_message_pair_in_logfile(log_filename, msg_start_of_section,
+                               msg_end_of_section):
+    """
+    Read the NEST Travis CI build log file and return 'True' in case both
+    messages are found in correct order. Return 'False' if only the first
+    message was found. Return 'None' in case the first or both messages
+    are not contained in the file.
+
+    :param log_filename:         NEST Travis CI build log file name.
+    :param msg_start_of_section: Message number string, e.g. "MSGBLD1234"
+    :param msg_end_of_section:   Message number string, e.g. "MSGBLD1234"
+    :return:                     True, False or None
+    """
```

Comment:
Very nice!

We usually use the [NumPy documentation style](https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt) for our Python docstrings. Could you please reformat the ones in this script for consistency reasons? Thanks!